packages/desktop-client/src/components/budget/BudgetTable.jsx

```diff
@@ -14,28 +15,56 @@ import { BudgetTotals } from './BudgetTotals';
 import { MonthsProvider } from './MonthsContext';
 import { findSortDown, findSortUp, getScrollbarWidth } from './util';
 
-class BudgetTableInner extends Component {
-  constructor(props) {
-    super(props);
-    this.budgetCategoriesRef = createRef();
+export function BudgetTable(props) {
+  const {
+    type,
+    prewarmStartMonth,
+    startMonth,
+    numMonths,
+    monthBounds,
+    dataComponents,
+    onSaveCategory,
+    onDeleteCategory,
+    onSaveGroup,
+    onDeleteGroup,
+    onReorderCategory,
+    onReorderGroup,
+    onShowActivity,
+    onBudgetAction,
+  } = props;
+
+  const budgetCategoriesRef = useRef();
+  const { grouped: categoryGroups } = useCategories();
+  const [collapsed = [], setCollapsedPref] = useLocalPref('budget.collapsed');
+  const [showHiddenCategories, setShowHiddenCategoriesPef] = useLocalPref(
+    'budget.showHiddenCategories',
+  );
+  const [editing, setEditing] = useState(null);
+
+  useEffect(() => {
+    const unlisten = listen('undo-event', () => {
+      // g dammit
```

Comment:
@MatissJanis Moved this from index budget file to `BudgetTable`. Do you think this is still applicable or can this be removed?

Reply:
Not sure. This is something from the time James was working on Actual.

I would remove it and see if anything breaks. If everything seems to be working OK - we can keep it removed.